addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/discovery/DSMRDiscoveryService.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.dsmr.internal.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterDescriptor;
+import org.openhab.binding.dsmr.internal.meter.DSMRMeterType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base class for discovery services.
+ *
+ * @author M. Volaart - Initial contribution
+ * @author Hilbrand Bouwkamp - Refactored code to detect meters during actual discovery phase.
+ */
+@NonNullByDefault
+public abstract class DSMRDiscoveryService extends AbstractDiscoveryService {
+    /**
+     * Timeout for Discovery timeout
+     */
+    private static final int DSMR_DISCOVERY_TIMEOUT_SECONDS = 120;
```

Comment:
2 minutes?

Reply:
I guess this can be shorter :smile: time between data can be 10 seconds, so detecting correct configuration could take 30 a 40 seconds. So rounded it up and multiplied it by 2. What would be a better timeout? 1 minute?

Reply:
Most bindings seem to settle for 10 seconds or 1 minute.
MqttServiceDiscoveryService even has 500 seconds. :frowning: